backend/src/score.ts

```diff
@@ -534,9 +534,22 @@ const scoreCity = (cityA: string|string[]|RequestField, cityB: string|string[]):
 const scoreLocationCode = (codeA: string|string[]|RequestField, codeB: string|string[]): number => {
     if (typeof(codeA) === 'string') {
         if (typeof(codeB) === 'string') {
-            return (codeA === codeB) ? 1 : minCodeScore;
+            if (codeA.length === 4) { codeA = `0${codeA}` }
```

Comment:
what if `codeB.length === 4` ?
I think it won't catch the next condition `codeA === codeB`

Reply:
the all code is coded specifically to insee data (dataset B).
codeB (insee code) has never 4 length (there are many hacks like that). Blind months and days will be coded YYYY0000

Reply:
Work is still in progress, all those test make performance very bad, i have to find more optimisations